academy/3-ibc/5-token-transfer.md

```diff
@@ -238,12 +187,103 @@ func (k Keeper) SendTransfer(
 }
 ```
 
+Take a look at the type [definition of a token packet](https://github.com/cosmos/ibc-go/blob/main/proto/ibc/applications/transfer/v2/packet.proto) before diving further into the code:
+
+```protobuf
+syntax = "proto3";
+
+package ibc.applications.transfer.v2;
+
+option go_package = "github.com/cosmos/ibc-go/v3/modules/apps/transfer/types";
+
+// FungibleTokenPacketData defines a struct for the packet payload
+// See FungibleTokenPacketData spec:
+// https://github.com/cosmos/ibc/tree/master/spec/app/ics-020-fungible-token-transfer#data-structures
+message FungibleTokenPacketData {
+    // the token denomination to be transferred
+    string denom = 1;
+    // the token amount to be transferred
+    string amount = 2;
+    // the sender address
+    string sender = 3;
+    // the recipient address on the destination chain
+    string receiver = 4;
+    // optional memo
+    string memo = 5;
+}
+```
+
+<HighlightBox type="note">
+
+Note that an optional _memo_ field was recently added to the packet definition. More details on the motivation, use cases and consequences will follow shortly.
```

Comment:
we can maybe link [the PR ](https://github.com/cosmos/ibc-go/pull/2305) which added the memo field

Reply:
> More details on the motivation, use cases and consequences will follow shortly.

Does this sort of message belong here? I think instead we should have an issue and maybe link in a TODO comment to flesh this out, WDYT?

Reply:
Fair point @chatton , I actually meant to link to [Adi's blog post](https://medium.com/the-interchain-foundation/moving-beyond-simple-token-transfers-d42b2b1dc29b) but I could not find it earlier

I find the PRs  I found on the memo field a bit hard to grasp for the beginner, so perhaps the blog post is more appropriate given the level of expertise with ibc-go of the dev portal reader

Reply:
I think linking the blog post is perfect!